geopandas/base.py

```diff
@@ -24,11 +24,11 @@ def is_geometry_type(data):
         return False
 
 
-def _delegate_binary_method(op, this, other, *args, **kwargs):
+def _delegate_binary_method(op, this, other, align, *args, **kwargs):
     # type: (str, GeoSeries, GeoSeries) -> GeoSeries/Series
     this = this.geometry
     if isinstance(other, GeoPandasBase):
-        if not this.index.equals(other.index):
+        if align and not this.index.equals(other.index):
             warn("The indices of the two GeoSeries are different.")
             this, other = this.align(other.geometry)
         else:
```

Comment:
you might need to check that the length matches in the `else` case now (because that is no longer guaranteed by the index equals check). Not fully sure what error you would otherwise get now (and we should also add a test for it)

Reply:
Ah, true. Will do.

Reply:
It is actually fine as it is now. We get `ValueError: Lengths of inputs do not match. Left: 3, Right: 4` from array.py where it is already tested. I would be just duplicating the check and error handling here. https://github.com/geopandas/geopandas/blob/01943827b005c89e28921ae8245da7127ecb8a09/geopandas/array.py#L537

Reply:
Ah, that's good of us ;)